app/models/job_proxy_dispatcher.rb

```diff
@@ -202,7 +202,7 @@ def assign_proxy_to_job(proxy, job)
   end
 
   def self.waiting?
-    Job.where(:state => 'waiting_to_start').exists?
+    Job.where(:state => 'waiting_to_start').exists? || InfraConversionJob.where.not(:state => ['finished', 'waiting_to_start']).exists?
```

Comment:
I don't like that this does two queries given how often this runs.

Instead of `||` can you use `.or()` something like this?

```
>> Job.where(:state => "waiting_to_start").or(Job.where(:type => "InfraConversionJob").where.not(:state => ["finished", "waiting_to_start"]))
  Job Load (0.9ms)  SELECT  "jobs".* FROM "jobs" WHERE ("jobs"."state" = $1 OR "jobs"."type" = $2 AND "jobs"."state" = 'running') LIMIT $3  [["state", "waiting_to_start"], ["type", "InfraConversionJob"], ["LIMIT", 11]]
  Job Inst Including Associations (0.0ms - 0rows)
=> #<ActiveRecord::Relation []
```

Reply:
Changed the query and added specs. Thanks @agrare